### DISCUSSION TITLE
How can I impose a boundary condition in a custom class without applying it to an operator?

### DISCUSSION THREAD

**Discussion by HannoSpreeuw** [zwicker-group/py-pde#255] [Q&A]
Thanks for building this awesome package.

I have used the [example SIR model](https://py-pde.readthedocs.io/en/latest/examples_gallery/pde_sir.html) as a starting point for my work.
From this example I got to understand how the boundary conditions are applied to the laplace operator: e.g. `s.laplace(self.bc)`.
But that is a boundary condition for the laplace operator applied to the field, **how do you specify a boundary condition for the field itself in the framework of the SIR example**?

And how do you do that when not all fields have the same boundary conditions, contrary to the SIR example?

From [this discussion](https://github.com/zwicker-group/py-pde/discussions/79) I understand that `bc_ops` can used for different boundary conditions for different fields. But that  discussion does not use the custom PDE class approach as in the example SIR model, so I am unable to map that to something I could use.

**Comment by david-zwicker** [comment]
The simplest option would be to supply different boundary conditions to different differential operators, i.e., when they apply to different fields. In the case of the SIR model, there are three calls to the `laplace` method, which in principle can receive different boundary conditions.

**Reply by HannoSpreeuw** [reply]
Thanks! I understand how you can impose a certain value of the laplace operator applied to the field as a boundary condition. 
But how can you impose a value of the field itself as a boundary condition when using a custom class?

**Reply by HannoSpreeuw** [reply]
Perhaps I interpret the [documentation about boundary conditions](https://py-pde.readthedocs.io/en/latest/manual/advanced_usage.html#boundary-conditions) incorrectly. When it says:
```
bc_lower = {"value": 4}
bc_upper = {"derivative": 2}
bc = [bc_lower, bc_upper]

grid = pde.UnitGrid([16])
field = pde.ScalarField(grid)
field.laplace(bc)
```
`value` and `derivative`, although they are applied as arguments of `field.laplace`, do not refer to the value and derivative of the laplacian of the field, but to the value and derivative of the field itself?
For me, this is counterintuitive.

**Reply by david-zwicker** [reply]
I see how this can be confusing, but this is the standard terminology. The boundary conditions that respectively set the value and the derivative of the field (before applying the operator) are often called [Dirichlet boundary conditions](https://en.wikipedia.org/wiki/Dirichlet_boundary_condition) and [Neumann boundary conditions](https://en.wikipedia.org/wiki/Neumann_boundary_condition).
